### ISSUE TITLE
supporting virtual machine installations with lower storage requirements

### ISSUE THREAD

**Issue by johnmaccormick** [DickinsonCollege/FarmData2#375]
As of January 2022 when I am creating this issue, the FarmData2
installation recommends a minimum of 30 GB of virtual disk space when
installing on a virtual machine. (This is a modest increment on the
amount of disk space recommended for a standard Ubuntu installation in
Jan 2022, which is 25 GB of available storage.) 

I'm wondering if it would be a worthy goal for the FarmData2 project
to support installations that have a smaller storage requirement. I
imagine that the majority of contributors do not have the luxury of
installing on a dedicated Linux box and are therefore using virtual
machines. Most contributors, especially students, will find it most
convenient to install a VM on their own laptop. This means freeing up
30 GB of space on a consumer laptop, which is a likely a significant
fraction of the laptop hard drive. For example, my two-year-old laptop
has a 237 GB hard drive, so the 30 GB requirement consumes about 12%
of the space. 

In reality, once a handful of VM snapshots have been taken, this
requirement will grow further, likely in the 40 GB range or more. (As
one data point to support this, I currently have a virtual machine
with three snapshots that together consume about 10 GB, which
suggests an average of 3 GB or more per snapshot.)

While it is always possible to free up tens of gigabytes on a laptop
by sacrificing other storage priorities, this requirement places an
obstacle in front of potential contributors that is perhaps
unnecessary.

Therefore, I'm suggesting that the FarmData2 community could debate
among the following options:
1. Do nothing
2. Add a little more explanation in the installation guidelines for
   people wishing to have a smaller storage footprint (e.g. via Linux
   Lite, which is already mentioned as a possibility in the
   installation instructions).
3. Switch to a different default Linux installation. In this scenario,
   the default FarmData2 setup and guidelines would use a minimal
   Linux setup. Contributors could still use a more fully-fledged
   Linux installation if desired, of course. But that would not be the
   default. The most carefully-tested and commonly-used installation
   would be a minimal installation.

**Comment by braughtg** [comment]
Good point and good ideas.  Happy to have any input suggestions as pull requests.  

Another fairly low cost option in the meantime is to store VMs on an external USB drive.

**Comment by johnmaccormick** [comment]
> Another fairly low cost option in the meantime is to store VMs on an external USB drive.

I found the performance unacceptably slow when I tested this option on an older external USB drive (a few years old). It is possible that the performance is good on more modern drives, especially USB-C perhaps. It would be interesting to see some reports from people who have tried a few different configurations.

**Comment by braughtg** [comment]
I don't have personal experience with this but one of my summer students ran from an external USB drive.  Perhaps she @IrisSC can comment on what the specs/configuration were.

**Comment by braughtg** [comment]
Some immediate thoughts I have on a minimum footprint install would be:
* Use Alpine Linux - this would likely require some extra prerequisite installations because it really is minimal. 
* the Theia ide docker image should be looked at - see issue #125 and #316. 
* the Cypress testing container is fairly large as well.

**Comment by IrisSC** [comment]
Hello,

I was able to use an external USB drive on my labtop. During the summer I was running it with Linux Lite and during this past fall semester I was using Ubuntu in the VM. 

The computer I was running it off of was a Lenovo Yoga 12(8GB of ram, and a HDD drive for storage), which at the time of writing this is 6.5 years old and a SanDisk Extreme Pro Portable SSD that has a USB 3 (probable type C, but not sure) that is at the time of writing 1 years old. The external drive that I have is on the upper ends in terms of cost, mostly because is has a lot of storage since I got it to use for other things as well. 

I have not run Ubuntu just on my laptop, since I am not sure how the speeds compare, but there didn't seem to be much of a difference running linux lite on the SSD drive that I have vs just on my computer. 

-Iris

**Comment by braughtg** [comment]
Current planning is that the FarmData2 developer install will be moving to a docker-based environment.  This will significantly reduce the overhead required and should also significantly improve the performance.
